naming/src/main/java/com/alibaba/nacos/naming/core/Instances.java

```diff
@@ -83,9 +84,10 @@ private String recalculateChecksum() {
             sb.append(",");
         }
 
-        if (MESSAGE_DIGEST != null) {
-            checksum =
-                new BigInteger(1, MESSAGE_DIGEST.digest((sb.toString()).getBytes(Charset.forName("UTF-8")))).toString(16);
+        if (MESSAGE_DIGEST_LOCAL.get() != null) {
```

Comment:
Here, you can consider holding the MessageDigest with a local variable to avoid getting again later

Reply:
> Here, you can consider holding the MessageDigest with a local variable to avoid getting again later

done